plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/SQLEditor.java

```diff
@@ -2258,25 +2261,7 @@ protected void doSetInput(IEditorInput editorInput) throws CoreException {
         }
         syntaxLoaded = false;
 
-        IEditorInput finalEditorInput = editorInput;
-        Runnable inputinitializer = () -> {
-            DBPDataSourceContainer oldDataSource = SQLEditor.this.getDataSourceContainer();
-            DBPDataSourceContainer newDataSource = EditorUtils.getInputDataSource(SQLEditor.this.getEditorInput());
-
-            if (oldDataSource != newDataSource) {
-                SQLEditor.this.dataSourceContainer = null;
-                SQLEditor.this.updateDataSourceContainer();
-            } else {
-                SQLEditor.this.reloadSyntaxRules();
-            }
-
-            {
-                DBPDataSourceContainer dataSource = EditorUtils.getInputDataSource(finalEditorInput);
-                SQLEditorFeatures.SQL_EDITOR_OPEN.use(Map.of(
-                    "driver", dataSource == null ? "" : dataSource.getDriver().getPreconfiguredId()
-                ));
-            }
-        };
+        Runnable inputinitializer = getEditorInputInitializer(editorInput);
```

Comment:
A continuation of the previous comment
```suggestion
        Runnable inputinitializer = () -> getEditorInputInitializer(editorInput);
```
But be careful as you might want to capture just the variable.